libs/full/async_cuda/src/cuda_future.cpp

```diff
@@ -80,10 +77,17 @@ namespace hpx { namespace cuda { namespace experimental { namespace detail {
             hpx::util::internal_allocator<>{}, stream);
     }
 
+#if defined(HPX_DEBUG)
+    static std::atomic<std::size_t> register_polling_count{0};
```

Comment:
In order to avoid initialization sequencing issues, would it make sense to wrap this as a function-local static instead? This would also have the benefit of not having any impact if not used.

Reply:
Good point. I suppose it's unlikely to be used like that, but better safe than sorry.